### PR TITLE
HPCC-16896 Specify ZAP report file name when creating the report.

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

### CODE REVIEW

**Comment by wangkx** [comment]
@RussWhitehead please review.

File: esp/services/ws_workunits/ws_workunitsService.cpp [hunk 1]
```diff
@@ -4650,7 +4650,22 @@ bool CWsWorkunitsEx::onWUCreateZAPInfo(IEspContext &context, IEspWUCreateZAPInfo
...
+        else
+        {
+            zipFileName.set(zipFName);
+            unsigned len = strlen(zipFName);
+            if (len < 5)
+                zipFileName.append(".zip");
+            else
+            {
+                char* tail = (char*) zipFName + len - 4;
+                if (!streq(tail, ".zip"))
```
Comment (RussWhitehead) [inline esp/services/ws_workunits/ws_workunitsService.cpp]: Should be strieq

File: esp/services/ws_workunits/ws_workunitsService.cpp [hunk 2]
```diff
@@ -4650,7 +4650,22 @@ bool CWsWorkunitsEx::onWUCreateZAPInfo(IEspContext &context, IEspWUCreateZAPInfo
...
+            zipFileName.append(nameStr.str()).append(".zip");
+        else
+        {
+            zipFileName.set(zipFName);
+            unsigned len = strlen(zipFName);
+            if (len < 5)
+                zipFileName.append(".zip");
+            else
+            {
+                char* tail = (char*) zipFName + len - 4;
```
Comment (RussWhitehead) [inline esp/services/ws_workunits/ws_workunitsService.cpp]: trivial, but should be "const char *"

**Review by RussWhitehead — CHANGES_REQUESTED** [review]

**Comment by wangkx** [comment]
@RussWhitehead please review my changes.

**Comment by RussWhitehead** [comment]
@wangkx Looks good

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged to 6.4.0

File: esp/services/ws_workunits/ws_workunitsService.cpp [hunk 3]
```diff
@@ -4650,7 +4650,22 @@ bool CWsWorkunitsEx::onWUCreateZAPInfo(IEspContext &context, IEspWUCreateZAPInfo
             addThorSlaveLogfile(cwu, winfo, folderToZIP.str());
 
         //Write out to ZIP file
-        zipFileName.append(nameStr.str()).append(".zip");
+        const char* zipFName = req.getZAPFileName();
+        if (!zipFName || !*zipFName)
+            zipFileName.append(nameStr.str()).append(".zip");
+        else
+        {
```
Comment (richardkchapman) [inline esp/services/ws_workunits/ws_workunitsService.cpp]: This code would be rather simpler/clearer using pathExtension (from jlib)

**Comment by richardkchapman** [comment]
@wangkx one comment

**Comment by wangkx** [comment]
@richardkchapman The code is using pathExtension() now.

**Pull request merged**